### ISSUE TITLE
openURL(_:) is deprecated

### ISSUE THREAD

**Issue by sammoore** [NativeScript/NativeScript#7448]
**Is your feature request related to a problem? Please describe.**
[`UIApplication#openURL(_:)`](https://developer.apple.com/documentation/uikit/uiapplication/1622961-openurl) is deprecated since iOS 10. It appears to be available still in iOS 12, but it will eventually need to be replaced and I could not find an issue for this.

This is used by `tns-core-modules/utils`'s `openUrl` [here](https://github.com/NativeScript/NativeScript/blob/5.4.2/tns-core-modules/utils/utils.ios.ts#L148).

**Describe the solution you'd like**
Switch to the Apple recommended [`UIApplication#open(_:options:completionHandler:)`](https://developer.apple.com/documentation/uikit/uiapplication/1648685-open)

**Describe alternatives you've considered**
- Continue using openURL until it is not available in a new iOS SDK release.
- Use `open(_:options:completionHandler:)` directly (not necessary yet / with current SDKs)

**Additional context**
I have not downloaded the headers for iOS 13 developer preview to see if `openURL(_:)` unavailable in the next version, but I assume it will be since Apple's own documentation still references it in [`UIApplication`'s overview](https://developer.apple.com/documentation/uikit/uiapplication#overview).

Also, there are implications that may have to surface up to the `tns-core-modules` API since the deprecated `openURL` is synchronous, but the replacement is asynchronous.

**Comment by NickIliev** [comment]
Although this is not actually a bug (yet) I will mark it as one so we that the issue can be prioritized. Marking as a `good-first-issue` - anyone interested in [contributing ](https://www.nativescript.org/contribute)and implementing [the newly introduced API](https://developer.apple.com/documentation/uikit/uiapplication/1648685-openurl?language=objc) is welcome.

**Comment by samtheprogram** [comment]
@NickIliev I'm willing to contribute here, but I'm not sure how to proceed regarding my comment on the API implications:

> Also, there are implications that may have to surface up to the tns-core-modules API since the deprecated openURL is synchronous, but the replacement is asynchronous.

In the meantime I'll review the contributing document you link to. Thanks for the quick response!

**Comment by sashok1337** [comment]
Any progress here? My app refuses to open links on iOS 13.3 with last NativeScript version, but in android everything works fine. Maybe apple finally removes openURL from API?

**Comment by ksharma20** [comment]
I guess this issue is already solved.
If yes then can you suggest some other good first issues, please? 
I am willing to contribute!

**Comment by awadeson** [comment]
Is this actually solved @ksharma20? 

Currently getting no response when clicking a link.

```
import { Utils } from '@nativescript/core';

saveToWallet(url: string): void {
    Utils.openUrl(url);
}

```
The following error is present when trying to use the block above:

```
BUG IN CLIENT OF UIKIT: The caller of UIApplication.openURL(_:) needs to migrate to the non-deprecated UIApplication.open(_:options:completionHandler:). Force returning false (NO).
```

Error only present on iOS 18.0 and above. Tested on iOS 17.5 and the link opened successfully.

**Comment by VepretskiyD** [comment]
The solution proposed here https://discord.com/channels/603595811204366337/603595811720527881/1292857447794606101 worked for me.
<img width="1780" alt="image" src="https://github.com/user-attachments/assets/c202adde-7130-49e8-8bf2-ab7a03cba277">

**Comment by rigor789** [comment]
https://github.com/NativeScript/NativeScript/commit/1cee35d4fb2b7eaaa7fbd6aa08b28e3e00dcc7dc

Should be released already in `@nativescript/core@8.8.4` I believe.

**Comment by sammoore** [comment]
Wow, it's been a minute. Glad to see this finally get addressed. Closing.
